bitcoin/contrib/test.sh

```diff
@@ -25,6 +25,8 @@ fi
 
 # Pin dependencies as required if we are using MSRV toolchain.
 if cargo --version | grep "1\.41"; then
+    # 1.0.157 uses syn 2.0 which requires edition 2018
+    cargo update -p serde --precise 1.0.156
```

Comment:
Should this be the first commit or we don't care about tests passing MSRV for the purpose of bisecting?

Reply:
No, we don't care, on multiple levels :)

* We don't care about MSRV for non-tip commits
* We don't care about contrib/test.sh passing for any commits, since this file is really for CI and is a PITA to run locally
* We *can't* care about pins working in old commits. Literally every commit in the repo, up to this one, now fail with 1.41, because the dependency changed from under us. So moving it back one or two commits wouldn't change that situation much.